alacritty_terminal/src/config/scrolling.rs

```diff
@@ -11,8 +11,10 @@ pub struct Scrolling {
     history: ScrollingHistory,
     #[serde(deserialize_with = "failure_default")]
     multiplier: ScrollingMultiplier,
+
+    // TODO: REMOVED
     #[serde(deserialize_with = "failure_default")]
-    pub auto_scroll: bool,
+    pub auto_scroll: Option<bool>,
```

Comment:
Just a little work, but could we make these `Deprecated<bool>` and clean up some of the code a bit?

Might save us some TODO comments, and might make the warning code simpler somehow.

Reply:
(in this case it should be something like `Removed<bool>` since it no longer influences functionality)

Reply:
How would it save us from the todo comments and make the warning code simpler?

Reply:
Just to expand a bit on this, if we wanted to hide this behind an abstraction we would have to put a bit more effort into it and write a macro for an annotation for example.

Something like `#[deprecated("Deprecation Message")]`.

But I don't think there's any simple way to implement this, so the current implementation should be less complex.

Reply:
Well just wrapping things in `Option` is a bit loose. This way we explicitly be parsing Removed or Deprecated things. We could handle them generically potentially.

But at the very least it's a wrapper for something like an Option that tells us we can check for `is_removed`.

Ideally the `print_deprecation_warnings` could be streamlined a bit too. But this is secondary to just having a reliable way to detect outdated configs.

In a perfect world, this would be a flag to serde deserialization that emits the warning without even needing to allocate the field I'd think, but that could be a later enhancement on the type.

Reply:
> Nothing tells me (other than a comment) that the field was removed

Well it wasn't removed entirely yet. It's just not required anymore. I'm also not sure how a type would be different from a comment. Unless specific behavior is associated to it, a named type is mostly the same.

> But at the very least it's a wrapper for something like an Option that tells us we can check for is_removed.

Not all deprecated options have been removed.

> In a perfect world, this would be a flag to serde deserialization that emits the warning without even needing to allocate the field I'd think, but that could be a later enhancement on the type.

First of all, I'm not very fond of rewriting serde, so that's not a realistic option. An annotation would be possible, however as previously mentioned that's more work than it is currently worth imho.

The much bigger problem I see is that there currently is no version attached to the deprecation warnings. So I have no way to tell when something deprecated should be completely removed. If I would write a macro to do all this for me, that's something I would include.

So yeah, enhancements here would be possible. But I wouldn't do that with a custom type, since it does not solve any of the fundamental problems imho. If there's a desire to fix these, a macro annotation is the way to go.

Reply:
Let's just keep this thing simple. There could be a state when an option could be completely removed and a new type aka `Removed<T>` will become useless. So using an Option<T> to solve this task makes sense, since its a stdlib feature that we're already heavily using among the code for similar things.